From 021a8171a09574a0d9389a531dadb4501c9f046d Mon Sep 17 00:00:00 2001
From: uni-dos <uni.dos12@outlook.com>
Date: Fri, 24 Apr 2026 11:11:54 -0700
Subject: [PATCH] do not install library

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index bf7f6a6..40d3469 100644
--- a/meson.build
+++ b/meson.build
@@ -189,7 +189,7 @@ imgui = library(
   gnu_symbol_visibility: 'hidden',
   dependencies: dependencies,
   include_directories: include_dirs,
-  install: true,
+  install: false,
   override_options: override_opts,
 )
 
-- 
2.53.0

