From ee5cf4b231848ae3e3d43c1a14115194e5d5d7f7 Mon Sep 17 00:00:00 2001
From: Clint Eschberger <clint@getsol.us>
Date: Thu, 23 Apr 2026 15:26:58 -0500
Subject: [PATCH] testsuite: disable panic tests

---
 Makefile.in        | 1 -
 testsuite/local.mk | 1 -
 2 files changed, 2 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index cb45c11..f951c8c 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -3594,7 +3594,6 @@ T = testsuite/misc.pl testsuite/backref-anchor.sh \
 	testsuite/mb-match-slash.sh testsuite/mb-y-translate.sh \
 	testsuite/missing-filename.sh testsuite/newline-dfa-bug.sh \
 	testsuite/normalize-text.sh testsuite/nulldata.sh \
-	testsuite/obinary.sh testsuite/panic-tests.sh \
 	testsuite/posix-backslash-warn.sh \
 	testsuite/posix-bracket-esc.sh testsuite/posix-char-class.sh \
 	testsuite/posix-mode-addr.sh testsuite/posix-mode-bad-ref.sh \
diff --git a/testsuite/local.mk b/testsuite/local.mk
index c350c49..20a3936 100644
--- a/testsuite/local.mk
+++ b/testsuite/local.mk
@@ -75,7 +75,6 @@ T =					\
   testsuite/normalize-text.sh		\
   testsuite/nulldata.sh			\
   testsuite/obinary.sh			\
-  testsuite/panic-tests.sh		\
   testsuite/posix-backslash-warn.sh	\
   testsuite/posix-bracket-esc.sh	\
   testsuite/posix-char-class.sh		\
-- 
2.53.0

