Adapt to our amdgpu kernel argument
diff --git a/lact-client/src/lib.rs b/lact-client/src/lib.rs
index 3132190..dc45a0a 100644
--- a/lact-client/src/lib.rs
+++ b/lact-client/src/lib.rs
@@ -127,8 +127,6 @@ impl DaemonClient {
     }
 
     request_plain!(get_system_info, SystemInfo, SystemInfo);
-    request_plain!(enable_overdrive, EnableOverdrive, String);
-    request_plain!(disable_overdrive, DisableOverdrive, String);
     request_plain!(generate_debug_snapshot, GenerateSnapshot, String);
     request_plain!(reset_config, RestConfig, ());
     request_with_id!(get_device_info, DeviceInfo, DeviceInfo);
diff --git a/lact-daemon/src/server.rs b/lact-daemon/src/server.rs
index 8f2a80d..ce44a0f 100644
--- a/lact-daemon/src/server.rs
+++ b/lact-daemon/src/server.rs
@@ -208,8 +208,6 @@ async fn handle_request<'a>(request: Request<'a>, handler: &'a Handler) -> anyho
             ok_response(handler.set_gpu_config(id, *config).await?)
         }
         Request::ProcessList { id } => ok_response(handler.process_list(id).await?),
-        Request::EnableOverdrive => ok_response(system::enable_overdrive().await?),
-        Request::DisableOverdrive => ok_response(system::disable_overdrive().await?),
         Request::GenerateSnapshot => ok_response(handler.generate_snapshot().await?),
         Request::ConfirmPendingConfig(command) => {
             ok_response(handler.confirm_pending_config(command)?)
diff --git a/lact-daemon/src/system.rs b/lact-daemon/src/system.rs
index 8645795..8d0a2b5 100644
--- a/lact-daemon/src/system.rs
+++ b/lact-daemon/src/system.rs
@@ -1,6 +1,6 @@
 pub mod power_profiles_daemon;
 
-use anyhow::{Context, anyhow, bail, ensure};
+use anyhow::{Context, anyhow};
 use lact_schema::{
     AmdgpuParamsConfigurator, BootArgConfigurator, GIT_COMMIT, InitramfsType, SystemInfo,
 };
@@ -13,36 +13,27 @@ use nix::sys::{
 use os_release::OsRelease;
 use std::{
     env,
-    fs::{self, File, Permissions},
-    io::{self, Write},
+    fs::{self},
+    io::{self},
     iter,
-    os::{fd::AsRawFd, unix::prelude::PermissionsExt},
-    path::{Path, PathBuf},
+    os::{fd::AsRawFd},
+    path::{Path},
     process::{self, Output},
     sync::{
-        LazyLock,
-        atomic::{AtomicBool, Ordering},
+        LazyLock
     },
 };
 use tokio::{process::Command, sync::Notify};
-use tracing::{debug, error, info};
-
-static OC_TOGGLED: AtomicBool = AtomicBool::new(false);
+use tracing::{debug, error};
 
 const PP_OVERDRIVE_MASK: u64 = 0x4000;
 pub const PP_FEATURE_MASK_PATH: &str = "/sys/module/amdgpu/parameters/ppfeaturemask";
 pub const BASE_MODULE_CONF_PATH: &str = "/etc/modprobe.d/99-amdgpu-overdrive.conf";
+pub const OVERDRIVE_ENABLED_PATH: &str = "/sys/module/amdgpu/parameters/overdrive_enabled";
 pub const DAEMON_VERSION: &str = env!("CARGO_PKG_VERSION");
 
 pub static IS_FLATBOX: LazyLock<bool> =
     LazyLock::new(|| env::var("FLATBOX_ENV").as_deref() == Ok("1"));
-static MODULE_CONF_PATH: LazyLock<PathBuf> = LazyLock::new(|| {
-    if *IS_FLATBOX {
-        Path::new("/run/host/root").join(BASE_MODULE_CONF_PATH.strip_prefix('/').unwrap())
-    } else {
-        PathBuf::from(BASE_MODULE_CONF_PATH)
-    }
-});
 
 pub async fn info() -> anyhow::Result<SystemInfo> {
     let version = DAEMON_VERSION.to_owned();
@@ -61,10 +52,12 @@ pub async fn info() -> anyhow::Result<SystemInfo> {
         |info| info.release().to_string_lossy().into_owned(),
     );
 
-    let amdgpu_overdrive_enabled = if let Ok(mask) = read_current_mask() {
+    let amdgpu_overdrive_enabled = if let Ok(true) = is_overdrive_enabled_flag() {
+        Some(true)
+    } else if let Ok(mask) = read_current_mask() {
         Some((mask & PP_OVERDRIVE_MASK) > 0)
     } else {
-        None
+    	None
     };
 
     let os_release = get_os_release().inspect_err(|err| error!("Could not detect distro: {err}"));
@@ -83,109 +76,15 @@ pub async fn info() -> anyhow::Result<SystemInfo> {
     })
 }
 
-pub async fn enable_overdrive() -> anyhow::Result<String> {
-    ensure!(
-        !OC_TOGGLED.load(Ordering::SeqCst),
-        "Overdrive support was already toggled - please reboot to apply the changes"
-    );
-
-    let current_mask = read_current_mask().context("Could not get current amdgpu feature mask")?;
-
-    let new_mask = current_mask | PP_OVERDRIVE_MASK;
-    if new_mask == current_mask {
-        return Err(anyhow!("Overdrive mask already enabled"));
-    }
-
-    let os_release = get_os_release()?;
-    let configurator = detect_amdgpu_configurator(&os_release).await?;
-    match configurator {
-        AmdgpuParamsConfigurator::Modprobe(initramfs_type) => {
-            let conf = format!("options amdgpu ppfeaturemask=0x{new_mask:X}");
-
-            let mut file =
-                File::create(&*MODULE_CONF_PATH).context("Could not open module conf file")?;
-            file.set_permissions(Permissions::from_mode(0o644))
-                .context("Could not conf file permissions")?;
-
-            file.write_all(conf.as_bytes())
-                .context("Could not write config")?;
-
-            if let Some(initramfs) = initramfs_type {
-                regenerate_initramfs(initramfs).await?;
-            }
-        }
-        AmdgpuParamsConfigurator::BootArg(BootArgConfigurator::RpmOstree) => {
-            run_command(
-                "rpm-ostree",
-                &[
-                    "kargs",
-                    &format!("--append-if-missing=amdgpu.ppfeaturemask=0x{new_mask:X}"),
-                ],
-            )
-            .await?;
-        }
-    }
-
-    OC_TOGGLED.store(true, Ordering::SeqCst);
-
-    // Returning a string only to maintain API compat
-    Ok(String::new())
-}
-
-pub async fn disable_overdrive() -> anyhow::Result<String> {
-    ensure!(
-        !OC_TOGGLED.load(Ordering::SeqCst),
-        "Overdrive support was already toggled - please reboot to apply the changes"
-    );
-
-    let os_release = get_os_release()?;
-    let configurator = detect_amdgpu_configurator(&os_release).await?;
-    match configurator {
-        AmdgpuParamsConfigurator::Modprobe(initramfs_type) => {
-            if Path::new(&*MODULE_CONF_PATH).exists() {
-                fs::remove_file(&*MODULE_CONF_PATH)
-                    .context("Could not remove module config file")?;
-
-                if let Some(initramfs) = initramfs_type {
-                    regenerate_initramfs(initramfs).await?;
-                }
-            } else {
-                bail!(
-                    "Overclocking was not enabled through LACT (file at {} does not exist)",
-                    MODULE_CONF_PATH.display(),
-                );
-            }
-        }
-        AmdgpuParamsConfigurator::BootArg(BootArgConfigurator::RpmOstree) => {
-            let current_mask =
-                read_current_mask().context("Could not get current amdgpu feature mask")?;
-            run_command(
-                "rpm-ostree",
-                &[
-                    "kargs",
-                    &format!("--delete-if-present=amdgpu.ppfeaturemask=0x{current_mask:X}"),
-                ],
-            )
-            .await?;
-        }
-    }
-
-    OC_TOGGLED.store(true, Ordering::SeqCst);
-
-    // Returning a string only to maintain API compat
-    Ok(String::new())
-}
-
-async fn regenerate_initramfs(initramfs_type: InitramfsType) -> anyhow::Result<()> {
-    info!("detected initramfs type {initramfs_type:?}, regenerating");
-    let result = match initramfs_type {
-        InitramfsType::Debian => run_command("update-initramfs", &["-u"]).await,
-        InitramfsType::Mkinitcpio => run_command("mkinitcpio", &["-P"]).await,
-        InitramfsType::Dracut => run_command("dracut", &["--regenerate-all", "--force"]).await,
-    };
-    result.context("Initramfs generation command failed")?;
-    Ok(())
+fn is_overdrive_enabled_flag() -> anyhow::Result<bool> {
+	let overdrive_enabled = fs::read_to_string(OVERDRIVE_ENABLED_PATH)?;
+	if overdrive_enabled.trim() != "0" {
+		Ok(true)
+	} else {
+		Ok(false)
+	}
 }
+ 
 
 fn read_current_mask() -> anyhow::Result<u64> {
     let ppfeaturemask = fs::read_to_string(PP_FEATURE_MASK_PATH)?;
diff --git a/lact-gui/src/app.rs b/lact-gui/src/app.rs
index 46ba0d4..df01f21 100644
--- a/lact-gui/src/app.rs
+++ b/lact-gui/src/app.rs
@@ -7,7 +7,6 @@ pub mod graphs_window;
 mod info_row;
 mod info_row_level;
 pub(crate) mod msg;
-mod overdrive_dialog;
 mod page_section;
 mod page_section_expander;
 pub(crate) mod pages;
@@ -21,7 +20,6 @@ use crate::{
     app::{
         about_dialog::{AboutDialog, AboutDialogMsg},
         gpu_selector::GpuSelector,
-        overdrive_dialog::{OverdriveDialog, OverdriveDialogMsg},
         preferences_dialog::{PreferencesDialog, PreferencesDialogMsg},
         process_monitor::{ProcessMonitorWindow, ProcessMonitorWindowMsg},
         profiles::{
@@ -102,7 +100,6 @@ pub struct AppModel {
     daemon_client: DaemonClient,
     graphs_window: relm4::Controller<GraphsWindow>,
     process_monitor_window: relm4::Controller<ProcessMonitorWindow>,
-    overdrive_dialog: relm4::Controller<OverdriveDialog>,
     preferences_dialog: relm4::Controller<PreferencesDialog>,
     about_dialog: relm4::Controller<AboutDialog>,
 
@@ -121,7 +118,7 @@ pub struct AppModel {
 
     settings_changed: BoolBinding,
 
-    system_info: SystemInfo,
+	system_info: SystemInfo,
     device_flags: Vec<DeviceFlag>,
     device_driver: String,
 }
@@ -292,17 +289,7 @@ impl AsyncComponent for AppModel {
                                         set_label: &fl!(I18N, "show-historical-charts"),
                                         connect_clicked => move |_| APP_BROKER.send(AppMsg::ShowGraphsWindow),
                                     },
-                                },
-
-                                add_top_bar = &adw::Banner {
-                                    #[watch]
-                                    set_revealed: model.system_info.amdgpu_overdrive_enabled == Some(false) && model.device_driver == "amdgpu",
-                                    set_title: &fl!(I18N, "amd-oc-disabled"),
-                                    set_use_markup: true,
-                                    set_button_label: Some(&fl!(I18N, "enable-amd-oc")),
-
-                                    connect_button_clicked => AppMsg::ShowOverdriveDialog,
-                                },
+                                },                             
 
                                 #[wrap(Some)]
                                 set_content = &gtk::ScrolledWindow {
@@ -445,10 +432,6 @@ impl AsyncComponent for AppModel {
             .launch(String::new())
             .forward(sender.input_sender(), |msg| msg);
 
-        let overdrive_dialog = OverdriveDialog::builder()
-            .launch((system_info.clone(), root.clone().upcast()))
-            .detach();
-
         let preferences_dialog = PreferencesDialog::builder()
             .launch((system_info.clone(), root.clone()))
             .detach();
@@ -472,7 +455,6 @@ impl AsyncComponent for AppModel {
             daemon_client,
             graphs_window,
             process_monitor_window,
-            overdrive_dialog,
             preferences_dialog,
             about_dialog,
             info_page,
@@ -586,9 +568,6 @@ impl AppModel {
             AppMsg::ShowAboutDialog => {
                 self.about_dialog.emit(AboutDialogMsg::Show);
             }
-            AppMsg::ShowOverdriveDialog => {
-                self.overdrive_dialog.emit(OverdriveDialogMsg::Show);
-            }
             AppMsg::SelectProfile {
                 profile,
                 auto_switch,
@@ -742,18 +721,6 @@ impl AppModel {
             AppMsg::DebugSnapshot => {
                 self.generate_debug_snapshot(root).await;
             }
-            AppMsg::EnableOverdrive => {
-                self.overdrive_dialog.emit(OverdriveDialogMsg::Loading);
-                let result = self.daemon_client.enable_overdrive().await;
-                self.overdrive_dialog.emit(OverdriveDialogMsg::Loaded);
-                result?;
-            }
-            AppMsg::DisableOverdrive => {
-                self.overdrive_dialog.emit(OverdriveDialogMsg::Loading);
-                let result = self.daemon_client.disable_overdrive().await;
-                self.overdrive_dialog.emit(OverdriveDialogMsg::Loaded);
-                result?;
-            }
             AppMsg::ResetConfig => {
                 self.daemon_client.reset_config().await?;
                 sender.input(AppMsg::ReloadData { full: true });
diff --git a/lact-gui/src/app/msg.rs b/lact-gui/src/app/msg.rs
index 8db14d0..552a0cc 100644
--- a/lact-gui/src/app/msg.rs
+++ b/lact-gui/src/app/msg.rs
@@ -25,9 +25,6 @@ pub enum AppMsg {
     DebugSnapshot,
     ShowPreferencesDialog,
     ShowAboutDialog,
-    ShowOverdriveDialog,
-    EnableOverdrive,
-    DisableOverdrive,
     ResetConfig,
     FetchProcessList,
     ReloadProfiles {
diff --git a/lact-gui/src/app/preferences_dialog.rs b/lact-gui/src/app/preferences_dialog.rs
index 6050795..77f26e8 100644
--- a/lact-gui/src/app/preferences_dialog.rs
+++ b/lact-gui/src/app/preferences_dialog.rs
@@ -108,21 +108,6 @@ impl relm4::Component for PreferencesDialog {
                 add = &adw::PreferencesGroup {
                     set_title: &fl!(I18N, "daemon"),
 
-                    adw::ActionRow {
-                        set_title: &fl!(I18N, "disable-amd-oc"),
-                        set_activatable: true,
-                        add_suffix = &gtk::Image {
-                            set_icon_name: Some("go-next-symbolic"),
-                        },
-
-                        #[watch]
-                        set_visible: model.system_info.amdgpu_overdrive_enabled.is_some(),
-
-                        connect_activated => move |_| {
-                            APP_BROKER.send(AppMsg::ShowOverdriveDialog);
-                        },
-                    },
-
                     adw::ActionRow {
                         set_title: &fl!(I18N, "reset-all-config"),
                         set_activatable: true,
diff --git a/lact-schema/src/request.rs b/lact-schema/src/request.rs
index 7c7c0c7..6fb7ccc 100644
--- a/lact-schema/src/request.rs
+++ b/lact-schema/src/request.rs
@@ -104,8 +104,6 @@ pub enum Request<'a> {
     ProcessList {
         id: &'a str,
     },
-    EnableOverdrive,
-    DisableOverdrive,
     GenerateSnapshot,
     ConfirmPendingConfig(ConfirmCommand),
     RestConfig,
