From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Mon, 18 Sep 2023 16:20:40 -0400
Subject: [PATCH 1/4] Don't spawn a new shell when starting

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 calamares.desktop    | 2 +-
 calamares.desktop.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/calamares.desktop b/calamares.desktop
index 9a1416e16..984b9970a 100644
--- a/calamares.desktop
+++ b/calamares.desktop
@@ -5,7 +5,7 @@ Name=Install System
 GenericName=System Installer
 Keywords=calamares;system;installer;
 TryExec=calamares
-Exec=sh -c "pkexec calamares"
+Exec=/usr/bin/calamares-gui
 Comment=Calamares — System Installer
 Icon=calamares
 Terminal=false
diff --git a/calamares.desktop.in b/calamares.desktop.in
index ed1d4def8..d13b60776 100644
--- a/calamares.desktop.in
+++ b/calamares.desktop.in
@@ -5,7 +5,7 @@ Name=Install System
 GenericName=System Installer
 Keywords=calamares;system;installer;
 TryExec=calamares
-Exec=sh -c "pkexec calamares"
+Exec=/usr/bin/calamares-gui
 Comment=Calamares — System Installer
 Icon=calamares
 Terminal=false
