--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -473,6 +473,8 @@
 else()
   message(STATUS "Building libbinaryen as shared library.")
   add_library(binaryen SHARED)
+  set_target_properties(binaryen PROPERTIES VERSION "0.0.${PROJECT_VERSION}")
+  set_target_properties(binaryen PROPERTIES SOVERSION 0)
   if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     # Disable interposition and resolve Binaryen symbols locally.
     add_link_flag("-Bsymbolic")
   endif()
 endif()
